docker/nzcvm_webapp_backend/app.py

```diff
@@ -0,0 +1,281 @@
+
+from flask import Flask, request, send_file, jsonify, abort, Response
+
+app = Flask(__name__)
+
+# --- Configuration ---
+# Define the path to the NZCVM script within the container
+NZCVM_SCRIPT_PATH = (
+    "/usr/local/lib/python3.12/site-packages/velocity_modelling/scripts/nzcvm.py"
+)
+
+
+# --- Helper Functions ---
+def create_config_file(
+    config_data: Dict[str, Union[str, float, int]], directory: str
+) -> str:
+    """Creates a temporary config file from a dictionary.
+    Assumes keys in config_data are already in the correct (uppercase) format.
+
+    Parameters
+    ----------
+    config_data : Dict[str, Union[str, float, int]]
+        A dictionary containing configuration key-value pairs.
+        Keys are expected to be uppercase strings representing configuration
+        parameter names, and values can be strings, floats, or integers.
+    directory : str
+        The path to the directory where the configuration file will be created.
+
+    Raises
+    ------
+    OSError
+        If there is an issue creating or writing to the config file.
+    """
+    config_content = [f"{key}={value}" for key, value in config_data.items()]
+    config_path = os.path.join(directory, "nzcvm.cfg")
+    with open(config_path, "w") as f:
+        f.write("\n".join(config_content))
+    print(f"Generated config file at: {config_path}")
+    print("Config file content:")
+    print("\n".join(config_content))  # Log the actual content written
+    return config_path
+
+
+def run_nzcvm_process(config_path: str, output_dir: str) -> bool:
+    """Runs the NZCVM command using subprocess.
+
+    Executes the NZCVM script to generate the velocity model based on
+    the provided configuration file, directing output to the specified directory.
+
+    Parameters
+    ----------
+    config_path : str
+        Path to the NZCVM configuration file. This file contains the
+        parameters required by the NZCVM script.
+    output_dir : str
+        Path to the directory where NZCVM output files should be saved.
+        The script will write its generated files into this location.
+
+    Returns
+    -------
+    bool
+        True if the NZCVM process completed successfully, False otherwise.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the NZCVM script or python interpreter is not found.
+    subprocess.CalledProcessError
+        If the NZCVM script returns a non-zero exit code.
+    subprocess.TimeoutExpired
+        If the NZCVM script execution exceeds the timeout.
+    """
+    # Construct the command to run the NZCVM script
+    command_args = [
+        "python",
+        NZCVM_SCRIPT_PATH,
+        "generate-velocity-model",
+        config_path,
+        "--out-dir",
+        output_dir,
+        "--output-format",
+        "HDF5",
+    ]
+
+    print(f"Running command: {' '.join(command_args)}")
+    try:
+        # Setting 10 min (600 seconds) timeout
+        result = subprocess.run(
```

Comment:
I assume you have to call as a subprocess because velocity_modellign could not be installed to the app env? Would be nicer to be able to call the function directly if possible, but this is fine if it's not possible.